src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -20,15 +20,19 @@ public class CompilerDataLogger
         private const string CommandLineEventName = "CommandLineInformation";
         private const string AssemblyReferencesEventName = "AssemblyReferencesInformation";
 
-        private readonly int ChunkSize;
-        private readonly string sha256;
+        private string Sha256 => context?.Hashes?.Sha256;
+
+        private readonly int chunkSize;
         private readonly string relativeFilePath;
+        private readonly IAnalysisContext context;
 
-        private static string s_sessionId;
+        internal static string s_sessionId;
```

Comment:
>internal [](http://example.com/codeflow?start=8&length=8)

please pull aside all the internal static properties (that are overridden by tests)  and group them together. 

we do this to improve the readability of code, i.e., that accessibility is a clue to the maintainer there is a special external caller (our tests) that update these values.

make sense? you are future-proofing this code to help whoever comes along when those tests fail.

Reply:
Got it!
thanks for the feedback!